### DISCUSSION TITLE
enh: Introduce Default State Configuration for Tools

### DISCUSSION THREAD

**Discussion by Simi5599** [open-webui/open-webui#7124] [General]
# Feature Request

**Is your feature request related to a problem? Please describe.**
Before Open WebUI v0.4.x, we could allow or disallow tools for a model and manually enable them via the chat interface. Now, if tools are allowed for a specific model, they are activated automatically, which can be frustrating.

**Describe the solution you'd like**
I believe we should implement a selector in the model configurator that allows setting the default state for specific tools. 
For example, if I have "Run code" and "Web search" tools, I want to be able to set it so that only the Web search tool is enabled in the chat by default instead of both. 
This way, I would have to manually turn off only the "Web search" tool if i don't need it.
Actually we can only disallow (disable via the model editor) the tools and for a non admin-use this makes the tool impossible to use.

**Comment by tjbck** [comment]
All tools are now available for all users by default, and you can set access control for each tool (instead of assigning them to a model), hope that helps! :)

**Reply by Simi5599** [reply]
Hi, thanks for the reply!

Perhaps I didn't express myself well due to my poor English: I wouldn't want to completely disable or hide the tools, but I would like there to be a selector that allows setting the default state of the tool (for example, normally off or normally on). In this case, we can consider that the tools before version 0.4 were all off by default.

**Comment by africsnail** [comment]
I have a related problem with the [autotool filter](https://openwebui.com/f/hub/autotool_filter). Now, if you want to use the filter, you need to enable the tools in the `workspace` > `models` section under each model on which you want to use the autotool filter. But after doing so, all the tools selected under the current model will be enabled by default in a new chat. This is because (at least as I get it) the per-model tool setting is essentially "enable by default" setting in the current version.

When using the autotool, you would want the tools **disabled by default**, as the autotool filter is supposed to save you the manual switching.

Other way to solve this would be to rewrite the autotool filter so it fetches all tools instead of fetching only the tools enabled in the model settings. I tried `available_tools = all_tools` and it seems to be working. But this of course creates another issue of not being able to exclude tools for the autotool filter to select from.
